src/ORM/DataObjectSchema.php

```diff
@@ -1046,6 +1088,16 @@ protected function getManyManyInverseRelationship($childClass, $parentClass)
      * @throws Exception
      */
     public function getRemoteJoinField($class, $component, $type = 'has_many', &$polymorphic = false)
+    {
+        return $this->getBelongsAndHasManyDetails($class, $component, $type, $polymorphic)['joinField'];
+    }
+
+    public function getHasManyComponentDetails(string $class, string $component)
+    {
+        return $this->getBelongsAndHasManyDetails($class, $component);
+    }
+
+    private function getBelongsAndHasManyDetails(string $class, string $component, string $type = 'has_many', &$polymorphic = false)
```

Comment:
```suggestion
    private function getBelongsToAndHasManyDetails(string $class, string $component, string $type = 'has_many', &$polymorphic = false): array
```
And update calls to this method

Reply:
Done